modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/MonadError.kt

```diff
@@ -55,27 +55,28 @@ interface MonadError<F, E> : ApplicativeError<F, E>, Monad<F> {
  *
  * typealias Impacted = Boolean
  *
- * object Nuke
- * object Target
- * class MissedByMeters(private val meters: Int) : Throwable("Missed by $meters meters")
+ * object Lettuce
+ * object Knife
+ * object Salad
+ * class InsufficientAmount(val quantityInGrams : Int) : Throwable("You need $quantityInGrams more grams of ingredient")
  *
- * fun <F> MonadThrow<F>.arm(): Kind<F, Nuke> = just(Nuke)
- * fun <F> MonadThrow<F>.aim(): Kind<F, Target> = just(Target)
- * fun <F> MonadThrow<F>.launchImpure(target: Target, nuke: Nuke): Impacted {
- *   throw MissedByMeters(5)
+ * fun <F> MonadThrow<F>.takeFoodFromRefrigerator(): Kind<F, Lettuce> = just(Lettuce)
+ * fun <F> MonadThrow<F>.getKnife(): Kind<F, Knife> = just(Knife)
+ * fun <F> MonadThrow<F>.launchImpure(tool: Knife, ingredient: Lettuce): Salad {
+ *   throw InsufficientAmount(5)
```

Comment:
What are 5 lettuces? Maybe a more discrete vegetable like a carrot might be a better example?

Edit: I noticed the quantity in grams. 5 g of lettuce seems like a pretty small quantity to me.

I really like the idea of replacing the nuclear war example with the cooking example.